bindata/network/ovn-kubernetes/common/008-script-lib.yaml

```diff
@@ -603,5 +612,56 @@ data:
         --acl-logging-rate-limit "{{.OVNPolicyAuditRateLimit}}" \
         ${gw_interface_flag} \
         ${ip_forwarding_flag} \
-        ${NETWORK_NODE_IDENTITY_ENABLE}
+        ${NETWORK_NODE_IDENTITY_ENABLE} \
+        ${ovn_v4_join_subnet_opt} \
+        ${ovn_v6_join_subnet_opt}
     }
+
+    start-ovnkube-control-plane()
+    {
+      local log_level=$1
+      local metrics_bind_ip=$2
+      local metrics_port=$3
+
+      if [[ $# -ne 3 ]]; then
+        echo "Expected three arguments but got $#"
+        exit 1
+      fi
+
+      ovn_v4_join_subnet_opt=
+      if [[ "{{.V4JoinSubnet}}" != "" ]] ; then
+        ovn_v4_join_subnet_opt="--gateway-v4-join-subnet {{.V4JoinSubnet}}"
+      fi
+      ovn_v6_join_subnet_opt=
+      if [[ "{{.V6JoinSubnet}}" != "" ]] ; then
+        ovn_v6_join_subnet_opt="--gateway-v6-join-subnet {{.V6JoinSubnet}}"
+      fi
+
+      ovn_v4_transit_switch_subnet_opt=
+      if [[ "{{.V4TransitSwitchSubnet}}" != "" ]] ; then
+        ovn_v4_transit_switch_subnet_opt="--cluster-manager-v4-transit-switch-subnet {{.V4TransitSwitchSubnet}}"
+      fi
+      ovn_v6_transit_switch_subnet_opt=
+      if [[ "{{.V6TransitSwitchSubnet}}" != "" ]] ; then
+        ovn_v6_transit_switch_subnet_opt="--cluster-manager-v6-transit-switch-subnet {{.V6TransitSwitchSubnet}}"
+      fi
+
+      echo "I$(date "+%m%d %H:%M:%S.%N") - ovnkube-control-plane - start ovnkube --init-cluster-manager ${K8S_NODE}"
+      exec /usr/bin/ovnkube \
+        --enable-interconnect \
+        --init-cluster-manager "${K8S_NODE}" \
+        --config-file=/run/ovnkube-config/ovnkube.conf \
+        --loglevel "${OVN_KUBE_LOG_LEVEL}" \
+        --metrics-bind-address "${metrics_bind_ip}:${metrics_port}" \
+        --metrics-enable-pprof \
+        --metrics-enable-config-duration \
+{{- if eq .ProductFlavor "managed" }}
+        --k8s-token-file=/var/run/secrets/hosted_cluster/token \
+        --node-server-privkey=/etc/pki/tls/metrics-cert/tls.key \
+        --node-server-cert=/etc/pki/tls/metrics-cert/tls.crt \
+{{- end }}
```

Comment:
So I guess that another option for things like metrics config and cert config is to set them as variables before calling this function, and check in this function their values and act accordingly.

Not strong opinion but what I propose would be play better with a future option of using pod env variables and would be consistent for all parameters (instead of passing arguments for some, and checking template variables for others)

Reply:
This logic is no more part of the implementation. Thanks for the suggestion though :)